### DISCUSSION TITLE
Running a whole lotta mods that worked before, added more and now I keep getting issues

### DISCUSSION THREAD

**Discussion by Snek-str** [TweakScale/TweakScale#274] [Support] [answered]
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/10149783/KSP.log)
^Logfile
I've been having troubles and multiple "Houston, we have a problem" pop ups, (**I'm assuming it's TweakScale causing these issues but B9PartSwitch had a fatal error too**) Game loads fine, its only when I try to exit is the issue, Same with the VAB and other buildings, VAB and SPH either don't let me enter, or don't let me leave, and when I do enter, I can place stuff fine, but I can't interact or delete them, It's either TweakScale or B9PartSwitch causing these issues, I hope you can help me fix these issues. I hope my saves weren't corrupted, and if they are, that it's repairable, Please get back to me ASAP, I don't care if you've got the solution yet or not, just comment stating that you've at least acknowledged this issue, Thanks!

**Comment by Snek-str** [comment]
PS: I do acknowledge I'm running outdated mods but they seem to be working fine, and I'm using ZeroMiniAVC to get rid of MiniAVC because it kept crashing my game

**Comment by Lisias** [comment]
Hi, @Tonkyboi1 !

No, it's not TwealScale causing issues. It's TweakScale detecting and yelling about issues. 99.5% of the time, it's something somewhere else triggering a KSP bug and screwing up TweakScale (and a lot of other add'ons, TS is not the only one being affected by some bugs, but it happens that it's one of the few ones complaining about them).

I'm inspecting your KSP.log to see what I find.

**Comment by Lisias** [comment, marked as answer]
And, yeah. Found it!

```
[ERR 14:10:34.357] AssemblyLoader: Exception loading 'ATW': System.Reflection.ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
  at (wrapper managed-to-native) System.Reflection.Assembly.GetTypes(System.Reflection.Assembly,bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0
  at AssemblyLoader.LoadAssemblies () [0x000e6] in <a1ca58b5ca7140639de29a81de5e3f32>:0

Additional information about this exception:

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.ModuleWeapon, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)

 System.TypeLoadException: Could not resolve type with token 0100006b (from typeref, class/assembly BDArmory.Modules.RocketLauncher, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.ModuleWeapon, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)
```

The module `ATW` is borking on being loaded because it's trying to load a very different version of BDA's module it was compiled against.

ATW is something from KFC (`KFC/Plugins/ATW`).

You need to try two things:

1) Update the BDA to the latest version, and see if the problem goes away.
2) If it doesn't, then you will need to remove KFC to keep playing, and then reach the KFC's maintainer's and ask for help.

Let me know if one of the options above worked for you.

**Reply by Snek-str** [reply]
![image](https://user-images.githubusercontent.com/87203360/205538064-83bf9983-2249-46b8-822a-7f22adc58d11.png)
^That still occurs but so far halfway into loading it seems OK
![image](https://user-images.githubusercontent.com/87203360/205538311-5266e8d6-bda5-4175-8905-27807576aaa8.png)
^Seems to get stuck on this now, Am I just able to remove that file it gets stuck on? Or will it throw errors?
`[LOG 20:45:31.421] PartLoader: Part 'BDArmory/Parts/F-86Launcher/F86Launcher/BahaF86Launcher' has no database record. Creating.
[LOG 20:45:31.421] [DragCubeSystem]: Drag cubes not found or cannot be read for part Part. Generating New drag cubes.
[LOG 20:45:31.424] DragCubeSystem: Creating drag cubes for part 'BahaF86Launcher'
[EXC 20:45:31.470] NullReferenceException: Object reference not set to an instance of an object
	BDArmory.Modules.ModuleWeapon.GetInfo () (at <dadbcdf9b9ad46c1a220dccbea06d76f>:0)
	PartLoader.CompilePartInfo (AvailablePart newPartInfo, Part part) (at <a1ca58b5ca7140639de29a81de5e3f32>:0)
	PartLoader+<CompileParts>d__56.MoveNext () (at <a1ca58b5ca7140639de29a81de5e3f32>:0)
	UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <12e76cd50cc64cf19e759e981cb725af>:0)
	UnityEngine.DebugLogHandler:LogException(Exception, Object)
	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
	UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
[LOG 20:47:41.157] DiscordRP: Plugin disable
[LOG 20:47:41.157] DiscordRP: Discord Shutdown`
^Snippet of logfile where it must have gotten stuck
PS: It just randomly started crashing when I tried again, Logfile provided here: [KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/10150286/KSP.log) , I have no clue why this started happening but it seems like usually whenever I delete cache files, it fixes.

**Reply by Lisias** [reply]
Install [TweakScale Companion for SMCE](https://github.com/net-lisias-ksp/TweakScaleCompanion_SMCE/releases), if you are using SMCE add'ons. These patches are **way** better than the hacks on the `Workarounds` folder.

I'm downloading the new `KSP.log` - and I find interesting that deleting the ConfigCache things get "fixed"….

**Reply by Lisias** [reply]
Found this:

```
[LOG 20:52:36.444] [BDArmory]=== Loading settings.cfg ===
[LOG 20:52:36.455] [BDArmory]: Error Loading Bullet Config | System.ArgumentNullException: Value cannot be null.
Parameter name: value
  at System.Number.ParseSingle (System.String value, System.Globalization.NumberStyles options, System.Globalization.NumberForma
  at System.Single.Parse (System.String s, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info)
  at System.Single.Parse (System.String s) [0x0000b] in <9577ac7a62ef43179789031239ba8798>:0
  at BDArmory.Bullets.BulletInfo.Load () [0x00095] in <dadbcdf9b9ad46c1a220dccbea06d76f>:0
```

Don't have a clue about the reason, you really need to reach BDA guys to seek further help - but I think some configurations failes are old, truncated or missing. Of perhaps on the wrong place?

Are you using any trick to get rid of the PD-Launcher?

**Reply by Snek-str** [reply]
I'm just about to go to sleep right now, and I'll be honest with ya, IDK what a PD-Launcher is, if ya explain it to me better, then I might know what it is (that is, when I read it when I wake up), Good night and thanks!

**Reply by Snek-str** [reply]
What is the PD-Launcher again? I've woken up now so you can reply whenever.

**Reply by Lisias** [reply]
PD-Launcher is a new Launcher Private Division is shoving in our arses on Steam. Yeah, the Steam Launcher launches a Launcher, and this 2nd level Launcher launches the game.

It's pretty useless now, but I concede that the stunt can be a good idea on the long run - as long they don't screw up things in the process.

The problem is that a lot of people didn't liked it, and don't want it (and some others got royally screwed by it!!) so they tried to get rid of the thing on the Steam Launcher's command line option. However, they changed the PWD of the launching to the PD-Launcher's directory, and so by replacing the PD-Launcher executable with the original KSP_x64.exe, you end up launching KSP from the wrong directory, and this broke both KSP and some add'ons.

Real mess.

But since you don't know what this crap is, you don't need to worry about it. Less one problem to handle.

**Reply by Snek-str** [reply]
I've noticed in the file the error points to, there's some empty values near the bottom
![image](https://user-images.githubusercontent.com/87203360/205698779-92ed5ad4-600a-48c7-9e2c-a9ac087d1143.png)
^Could that possibly be the cause? As it did say `Value cannot be null.`

**Reply by Lisias** [reply]
Perhaps - but I really don't know.

This is something to be asked to the BDA's maintainers - it's their kid, not mine!! :P 

I don't even know what should be put there. **PERHAPS** you can try to delete these empty values to see what happens? (make backups first!)

**Reply by Snek-str** [reply]
Seems to fix the crashing issue, I have asked the BDA Maintainers but they haven't gotten back to me for the past few hours, you've gotten back to me quickly so far so I trust you, I have deleted the values that are empty, now if I need any of these values in the future, I will just restore the value I need and map it to a key. I still need to fix the getting stuck on this part of loading sequence:
![image](https://user-images.githubusercontent.com/87203360/205706203-637343b4-6796-4d2f-95eb-d1bba65c848e.png)
If you want to help with that too, tell me and I can provide the new latest log file, or I could give you a break and ask the BDA guys instead, Thanks for all your help!
PS: Noticed on most of BDArmory issues github page, 1/3 issues are left unsolved, those solved, 1/5 of them solved themselves.

**Reply by Lisias** [reply]
The problem on trying to guess things on BDA is that I really don't know what I'm doing - sometimes, a workaround that appears to fix something can lead to a worst problem later. **BUT**, on this specific case of the settings, I'm guessing that you should be alright.

but keep backups, just in case. I may be wrong.

About this part… Do you have a reason to keep it? You can try to rename the file `BDArmory/Parts/F-86Launcher/F86Launcger.cfg` to `BDArmory/Parts/F-86Launcher/F86Launcger.cfg.bkp` to get rid of the problem - but this will remove this part from your game. Not to mention that perhaps this part is just a screaming victim, not the perpetrator of the crime, and so we may just allowing the criminal to seek another victim.

Let's try this: rename the file and then send me a new `KSP.log`. Let's see what happens. (and, as always, don't risk valuable savegames on these tests, backup everything. twice.)

**Reply by Snek-str** [reply]
Yeah... About that... It seems as though it randomly started crashing again... For the same reason even after it started working and I removed the nulls
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/10157431/KSP.log)

**Reply by Lisias** [reply]
Oukey, temos um bug "sério" no BDA. É o mesmo problema lá de cima:

```
[LOG 13:50:01.027] [BDArmory]=== Loading settings.cfg ===
[LOG 13:50:01.055] [BDArmory]: Error Loading Bullet Config | System.ArgumentNullException: Value cannot be null.
Parameter name: value at System.Number.ParseSingle (System.String value, System.Globalization.NumberStyles options, System.Globalization.NumberFormatInfo numfmt) [0x
  at System.Single.Parse (System.String s, System.Globalization.NumberStyles style, System.Globalization.NumberFormatInfo info) [0x00000] in <957
  at System.Single.Parse (System.String s) [0x0000b] in <9577ac7a62ef43179789031239ba8798>:0
  at BDArmory.Bullets.BulletInfo.Load () [0x00095] in <dadbcdf9b9ad46c1a220dccbea06d76f>:0
```

Ficar editando o Settings.cfg o tempo todo antes de levantar o KSP é um pé-no-saco, os caras do BDA precisam resolver isso eles mesmos - e eu diria que bem rápido...

**Reply by Snek-str** [reply]
Ok, I've fixed the startup crashing issue with BDA+ but now I get the "Houston, we have a problem" again
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/10157805/KSP.log)
Can ya look into that for me please? Thanks for all your support and help

**Reply by Lisias** [reply]
Ouch… I had answered in PT-BR by accident, sorry! (I'm on working hours and I'm task-switching!)

**Reply by Lisias** [reply]
It's the "ATW" again:

```
[ERR 14:56:54.887] AssemblyLoader: Exception loading 'ATW': System.Reflection.ReflectionTypeLoadException: Exception of type 'System.Reflection.ReflectionTypeLoadException' was thrown.
  at (wrapper managed-to-native) System.Reflection.Assembly.GetTypes(System.Reflection.Assembly,bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0
  at AssemblyLoader.LoadAssemblies () [0x000e6] in <a1ca58b5ca7140639de29a81de5e3f32>:0

Additional information about this exception:

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.ModuleWeapon, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)

 System.TypeLoadException: Could not resolve type with token 0100006b (from typeref, class/assembly BDArmory.Modules.RocketLauncher, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.ModuleWeapon, BDArmory, Version=1.3.4.0, Culture=neutral, PublicKeyToken=null)
```

You are on a crossroad on this one. If you are sure you are using the latest BDA, then you need to reach KFC maintainers for help - they need to publish a new Release of KFC compiled againts the newest BDA.

Otherwise, perhaps it's the BDA the one in need to be updated.

In a way or another, this is all I can infer without digging on the code of these two thinges - and I kinda of out-of-time to do such for some time - to tell you the true, I'm neglecting my own add'ons these days due Real Life™ … :/

**Reply by Snek-str** [reply]
Wrote a long thing just to have my PC Crash mid- writing, TL;DW: I don't know which mod adds KFC, Can you link me to them or their support page or at least tell me?

**Reply by Lisias** [reply]
Are you using CKAN or CurseInstaller?

**Reply by Lisias** [reply]
AH, I think I found it!! 

Kerbal Fuel Carrier. A pretty outdated add'on, I think it was replaced by [AARS](https://spacedock.info/mod/1167/Automated%20Aerial%20Refueling%20System).

Check this thread:

https://forum.kerbalspaceprogram.com/index.php?/topic/179515-191automated-aerial-refueling-system-continued/

The installation instructions mentions KFC:

> INSTALLATION: Delete the existing AARS and KFC folder in your own GameData, then unzip GameData into the game directory.

**Reply by Lisias** [reply]
Humm.. Before doing that… please ZIP the KFC folder and attach it to this thread, I want to check it. The true is that I'm blindly guessing what's KFC is...

**Reply by Snek-str** [reply]
File too large, Uploaded it to mega, Here is the link: https://mega.nz/file/MSNwSSZZ#MDiflPHcxnjgHiNYHaSozzK8yVy4mW-L3T_YNxCiIc8
I've searched Kerbal Fuel Carrier and Aircraft Carrier Accessories came to mind so I downloaded it and sure enough, in the Gamedata folder in the mod, KFC, KFC is Aircraft Carrier Accessories.

**Reply by Lisias** [reply]
Probably faster reply I ever did! 40 seconds! :)

**Reply by Snek-str** [reply]
PS: I don't use CKAN or CurseInstaller, If those 2 things are automated installers or mod managers, then I just install mods manually.

**Reply by Lisias** [reply]
I'm downloading the thing, but I think I finally found it:

http://forum.kerbalspaceprogram.com/index.php?/topic/152916-121aircraft-carrier-accessories/

Just don't ask me how Aircraft Carrier Accessories became KFC...

**Reply by Lisias** [reply]
Bingo. It's really ACA:

![Screen Shot 2022-12-05 at 20 07 53](https://user-images.githubusercontent.com/64334/205762742-9bf84450-7961-40e9-bd9a-63a31ea6f370.png)
![Screen Shot 2022-12-05 at 20 07 39](https://user-images.githubusercontent.com/64334/205762770-ff5042b4-055e-4ca9-89cf-d2192efd8d0b.png)

And you can download it from here: https://spacedock.info/mod/1068/Aircraft%20Carrier%20Accessories#changelog

However… the biggest file has only 25 megabytes, I don't know why your version have 50…

**Reply by Snek-str** [reply]
Re-Downloaded it, only to find my game is crashing on startup due to BDA again, but for a different reason this time
`[LOG 17:15:02.442] [BDArmory.BulletInfo]: Parsing default bullet definition from BD_Bullets [EXC 17:15:02.471] ArgumentNullException: Field 'DisplayName' is missing. Parameter name: DisplayName BDArmory.Bullets.BulletInfo.ParseField (ConfigNode node, System.String field, System.Type type) (at <b7a051ec31ab4c2f8b9d77a5ab9d7512>:0) BDArmory.Bullets.BulletInfo.Load () (at <b7a051ec31ab4c2f8b9d77a5ab9d7512>:0) BDArmory.UI.BDArmorySetup.Start () (at <b7a051ec31ab4c2f8b9d77a5ab9d7512>:0) UnityEngine.DebugLogHandler:LogException(Exception, Object) ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object) UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)`
^Snippet of logfile where error occurred, Need the full logfile? Notify me if so

**Reply by Snek-str** [reply]
Traced back the file next to ACA.dll, ACA is an extension for (or includes?) Advanced Tactical Weapons (ATW), OR ATW and ACA is an extension to KFC (making me hungry now)

**Reply by Lisias** [reply]
Hi! Send me the full KSP.log. I want to see the whole stacktrace of the thing.

We may be going to the wrong way - perhaps you should downgrade BDA to a version more or less the same date as the ACA and ATW files?

**Reply by Snek-str** [reply]
I have re-installed BDA+ and most of the add-ons for it, I am very sorry for the late response, I will try to get back to you quicker now, Heres the logfile
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/10166971/KSP.log)
Please help, Thanks!

**Reply by Snek-str** [reply]
Any reply yet? Its just been about a day now and my game still crashes

**Reply by Lisias** [reply]
We are on a dead lock. ATW still refuses to work with BDA.

```
[ERR 07:40:39.235] AssemblyLoader: Exception loading 'ATW': System.Reflection.ReflectionTypeLoadException: Exception of
  at (wrapper managed-to-native) System.Reflection.Assembly.GetTypes(System.Reflection.Assembly,bool)
  at System.Reflection.Assembly.GetTypes () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0
  at AssemblyLoader.LoadAssemblies () [0x000e6] in <a1ca58b5ca7140639de29a81de5e3f32>:0

Additional information about this exception:

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.Mod

 System.TypeLoadException: Could not resolve type with token 0100006b (from typeref, class/assembly BDArmory.Modules.Roc

 System.TypeLoadException: Could not resolve type with token 0100004f (from typeref, class/assembly BDArmory.Modules.Mod
```

I had confirmed that you are using the latest BDAc (by the Assembly version, 1.3.4.0) so there's no other culprit possible but ATW itself. Something changed on the latest BDAc that broke Assemblies compiled with an older BDA (possibibly ATW was compiled agains the original BDA, not the BDAc).

You are between a rock and a hard place: you need to downgrade BDA (and so everything that relies on it) until the damned thing works, or you need to wave off KFC/[AARS](https://spacedock.info/mod/1167/Automated%20Aerial%20Refueling%20System).

As it appears, there's no point on asking for help from the BDAc guys, as this is not something they can fix (but it would be nice if they had explained it to you, it would had saved a lot of time for both of us).

There's a really compelling reason to have KFC/AARS installed? The easiest way our of the problem, now that we have enough information, is to just remove it from your rig...

**Reply by Snek-str** [reply]
Update on situation, Noticed that KSP is randomly crashing whenever it wants, so I will probably stop reporting crashes to you as they only seem to occur on random, Just got rid of KFC, Goodbye Kentucky Fried Chicken or whatever its supposed to stand for, The game launches just fine but I'm having issues with Sandbox mode, as when I place down any ship part, I cannot interact with it in anyway, nor can I leave the building and I have to Alt F4 out of the game as nothing else will work, Any fix for this? And I'm sorry about all this trouble, It is crashing at what appears to be randomly, Installed BDAc instead of BDA+ to fix an issue with fields in logfiles not found added by BDA+, Just started the game again, Ill update you on another reply as this reply is getting too long

**Reply by Snek-str** [reply]
![image](https://user-images.githubusercontent.com/87203360/206242514-a5503184-65a5-49da-b607-9d3a886ee53d.png)
^First error I encounter, So far so... **_alright_**, Continuing to write comment and watch game for errors.

Seems to be the only error I encounter, testing the functions in all modes see which ones are broken and if they all work.
Sandbox - Ah, the gamemode where I'm having most issues with, It works great! I made this yacht but **_IDK how to get it in the water if ya can help me with that!_**
![image](https://user-images.githubusercontent.com/87203360/206248359-3a2a00ee-1580-4328-bec8-19f3322d84bf.png)

Science - Opened a few buildings and exited them fine, Built a "little" rocket called "The Abomi-Nation", Forgot the parachute again and engine blew up? Catastrophic failure screen closes instantly but I don't need it, Works fine other than that
![image](https://user-images.githubusercontent.com/87203360/206245021-3aa6515a-3709-451b-bea3-b192bfe1469a.png)

Career - Works fine, Made this little goblin and called it "The Abominator", lets just say I forgot the parachutes, seems exiting to main menu works fine.
![image](https://user-images.githubusercontent.com/87203360/206243925-318fe510-9a6a-4954-b968-a2752078c896.png)

All in all, it's working well! If ya wanna help me with that B9PartSwitch issue, just let me know, Also can ya tell me how to efficiently get a boat into the water without moving mods? Thanks for all your help!

**Reply by Lisias** [reply]
The problem with B9-PartSwitch is usually a bad patching. Looking on the ModuleManager Patch log will show who had patched ```fft-atmosphere-scoop-1``` and then someone can try a fix.

FFT is Far Future Technologies, and I found the "source" code for this part [here](https://github.com/post-kerbin-mining-corporation/FarFutureTechnologies/blob/b93b85c0081a2fb439ab3adbd147c342bb082ba0/GameData/FarFutureTechnologies/Parts/Resources/fft-atmosphere-scoop-1.cfg). There's no support for B9-PS on FFT, so I don't think the FFT's maintainer will be able to help you neither.

I think you should reach the B9PS's maintainers for help on this specific issue. Try [here](https://forum.kerbalspaceprogram.com/index.php?/topic/140541-1112-b9partswitch-v2180-march-17/page/8/) or [here](https://github.com/blowfishpro/B9PartSwitch/issues).

About launching your ship, try [Vessel Mover](https://forum.kerbalspaceprogram.com/index.php?/topic/183963-ksp-1122-vesselmover-continued-sep-2021/). There's also this [Water Launch Sites ](https://forum.kerbalspaceprogram.com/index.php?/topic/183937-1122-water-launch-sites-2021-09-11/)one, looks promising (but I never used it). I think you should install both, the extra Launch Sites for making your life easier, and the Vessel Mover for ~~cheatin~~… I mean, fine tunning your game! :)

**Reply by Snek-str** [reply]
These seem to be the only errors I found in MMPatch.log:
`[ERR 11:48:37.741] name specifier detected on insert node (not a patch): TweakScale/Workarounds/SMI_AddOns/PART[aftdeck,aftlower]:FINAL
`
`[ERR 11:48:37.742] pass specifier detected on insert node (not a patch): TweakScale/Workarounds/SMI_AddOns/PART[aftdeck,aftlower]:FINAL`

PS: I start my game again and now the error is different?
![image](https://user-images.githubusercontent.com/87203360/206528021-f8ce1010-d8ab-45d9-891d-16893923b3c0.png)
I will get support from B9PS, Thank you for all your help and have a great day! :)

**Reply by Lisias** [reply]
@Tonkyboi1 ,

> These seem to be the only errors I found in MMPatch.log:
> 
> `[ERR 11:48:37.741] name specifier detected on insert node (not a patch): TweakScale/Workarounds/SMI_AddOns/PART[aftdeck,aftlower]:FINAL ` `[ERR 11:48:37.742] pass specifier detected on insert node (not a patch): TweakScale/Workarounds/SMI_AddOns/PART[aftdeck,aftlower]:FINAL`

UGH… Well, just add a `@` before the `PART` and this will be fixed. I can swear I had fixed this before, but as we can see, I didn't.

Additionally, you will probably prefer to remove the Workaround and install [TweakScale Companion for SMCE](https://github.com/net-lisias-ksp/TweakScaleCompanion_SMCE). **WAY** better patches! (but you will need to replace a file on the SMCE's folders, because I couldn't manage to fix that one with patches - I'm not exactly proud of this hack, but it's what I could think at that time).

(sorry being late, things are somewhat "interesting" around here where I live)
